Smallrye config and microprofile config version conflict causes build failure with java modules

### Describe the bug

If you have a project which has a module-info file and also requires smallrye config, the build will fail starting with Quarkus version 3.32.0

### Cause:
In Quarkus version 3.32.0 the smallrye config version was bumped from 3.15.1 to 3.16.0 (#52245).
The microprofile config version was however not aligned to 3.1.1-RC1, which smallrye version 3.16.0 uses.

Smallrye added a module-info file in this realease (smallrye/smallrye-config#1459). Including required microprofile config, specific `org.eclipse.microprofile.config`.
Between Microprofile config version 3.1 and 3.1.1-RC1, the module name for Microprofile config was changed from the default value `microprofile.config.api` to `org.eclipse.microprofile.config` (microprofile/microprofile-config#810).

Since Quarkus still uses Microprofile config version 3.1 during build only the module `microprofile.config.api` is avaible, causing the build to fail, since it can not find the new module name `org.eclipse.microprofile.config`.

### Workaround:

Manually specific the Microprofile version before quarkus in the dependencymanagement:
```xml
    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>org.eclipse.microprofile.config</groupId>
                <artifactId>microprofile-config-api</artifactId>
                <version>3.1.1-RC1</version>
            </dependency>

            <dependency>
                <groupId>io.quarkus.platform</groupId>
                <artifactId>quarkus-bom</artifactId>
                <version>3.32.3</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>
    </dependencyManagement>
```

### Potential Fix:

Bump Microprofile Config version to 3.1.1-RC1. After Building and testing Quarkus locally, this seams to work. If this is an acceptable solution, I would be happy to make a PR.

### Expected behavior

Building a project with java module and using smallrye config should work.

### Actual behavior

Build fails with error:
```
ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.13.0:compile (default-compile) on project example: Compilation failure
[ERROR] Modul nicht gefunden: org.eclipse.microprofile.config
```

### How to Reproduce?

See: https://github.com/DerFrZocker/examples/tree/quarkusio/quarkus/config-version-mismatch

### Output of `uname -a` or `ver`

Linux marvin 6.17.0-19-generic #19~24.04.2-Ubuntu SMP PREEMPT_DYNAMIC Fri Mar  6 23:08:46 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk 25.0.2 2026-01-20 LTS OpenJDK Runtime Environment Temurin-25.0.2+10 (build 25.0.2+10-LTS) OpenJDK 64-Bit Server VM Temurin-25.0.2+10 (build 25.0.2+10-LTS, mixed mode, sharing)

### Quarkus version or git rev

3.32.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

pache Maven 3.9.11 (3e54c93a704957b63ee3494413a2b544fd3d825b) Maven home: /home/marvin/.local/share/JetBrains/Toolbox/apps/intellij-idea-ultimate/plugins/maven/lib/maven3 Java version: 25.0.2, vendor: Eclipse Adoptium, runtime: /home/marvin/.jdks/temurin-25.0.2 Default locale: de_DE, platform encoding: UTF-8 OS name: "linux", version: "6.17.0-19-generic", arch: "amd64", family: "unix"

### Additional information

_No response_

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -36,7 +36,7 @@
         <hdrhistogram.version>2.2.2</hdrhistogram.version><!-- keep in sync with micrometer -->
         <google-auth.version>0.22.0</google-auth.version>
         <graphql-java.version>25.0</graphql-java.version> <!-- keep in sync with smallrye-graphql -->
-        <microprofile-config-api.version>3.1</microprofile-config-api.version>
+        <microprofile-config-api.version>3.1.1-RC1</microprofile-config-api.version> <!-- keep in sync with smallrye-config -->
         <microprofile-health-api.version>4.0.1</microprofile-health-api.version>
         <microprofile-context-propagation.version>1.3</microprofile-context-propagation.version>
         <microprofile-fault-tolerance-api.version>4.1.2</microprofile-fault-tolerance-api.version>
@@ -46,7 +46,7 @@
         <microprofile-lra.version>2.0.2</microprofile-lra.version>
         <microprofile-openapi.version>4.1.1</microprofile-openapi.version>
         <smallrye-common.version>2.18.1</smallrye-common.version>
-        <smallrye-config.version>3.17.2</smallrye-config.version>
+        <smallrye-config.version>3.17.2</smallrye-config.version> <!-- keep in sync with microprofile-config-api -->
         <smallrye-health.version>4.3.0</smallrye-health.version>
         <smallrye-modules.version>1.0.beta1</smallrye-modules.version>
         <smallrye-open-api.version>4.3.3</smallrye-open-api.version>

--- a/extensions/packaging/modular/deployment/src/main/java/io/quarkus/modular/deployment/ModularitySteps.java
+++ b/extensions/packaging/modular/deployment/src/main/java/io/quarkus/modular/deployment/ModularitySteps.java
@@ -91,8 +91,6 @@ public final class ModularitySteps {
             CurateOutcomeBuildItem curateOutcome) {
         // TODO: migrate these to their relevant extensions
         return List.of(
-                new AddDependencyBuildItem("org.eclipse.microprofile.config", "io.smallrye.config",
-                        Modifier.Set.of(Modifier.SERVICES)),
                 // todo: this one must be READ and LINKED because an ArC synthetic bean requires it
                 new AddDependencyBuildItem("io.netty.transport", "io.quarkus.netty",
                         Modifier.Set.of(Modifier.SERVICES, Modifier.READ, Modifier.LINKED)),

